network-mux/src/Network/Mux.hs

```diff
@@ -406,8 +410,17 @@ monitor tracer timeout jobpool egressQueue cmdQueue muxStatus =
           throwIO e
         EventJobResult (DemuxerException e) -> do
           traceWith tracer (MuxTraceState Dead)
-          atomically $ writeTVar muxStatus $ MuxFailed e
-          throwIO e
+          r <- atomically $ do
+            size <- JobPool.readGroupSize jobpool MiniProtocolJob
+            case size of
+              0  | Just (MuxError MuxBearerClosed _) <- fromException e
+                -> writeTVar muxStatus MuxStopped
+                >> return True
+              _ -> writeTVar muxStatus (MuxFailed e)
+                >> return False
+          if r
+            then return ()
+            else throwIO e
```

Comment:
:+1: no more 'MuxError BearerClosed' questions!

Reply:
but only in the `p2p` branch. Clean close needs to await for all mini-protocols to terminate and we don't have such a component in the current system.  It wouldn't be difficult to implement though.